examples/plot_nonstationary_error.py

```diff
@@ -0,0 +1,205 @@
+
+Here, we show an example in which we identify terrain-related non-stationarities for a DEM difference of Longyearbyen glacier.
+We quantify those non-stationarities by `binning <https://en.wikipedia.org/wiki/Data_binning>`_ robustly
+in N-dimension using :func:`xdem.spstats.nd_binning` and applying a N-dimensional interpolation
+:func:`xdem.spstats.interp_nd_binning` to estimate a numerical function of the measurement error and derive the spatial
+distribution of elevation measurement errors of the difference of DEMs.
+
+**Reference**: `Hugonnet et al. (2021) <https://doi.org/10.1038/s41586-021-03436-z>`_, applied to the terrain slope
+and quality of stereo-correlation (Equation 1, Extended Data Fig. 3a).
+"""
+# sphinx_gallery_thumbnail_number = 8
+import matplotlib.pyplot as plt
+import numpy as np
+import xdem
+import geoutils as gu
+
+# %%
+# We start by loading example files including a difference of DEMs at Longyearbyen glacier, the reference DEM later used to derive
+# several terrain attributes, and the outlines to rasterize a glacier mask.
+# Prior to differencing, the DEMs were aligned using :class:`xdem.coreg.NuthKaab` as shown in
+# the :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py` example. We later refer to those elevation differences as *dh*.
+
+ref_dem = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"))
+ddem = xdem.DEM(xdem.examples.get_path("longyearbyen_ddem"))
+glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+mask_glacier = glacier_outlines.create_mask(ddem)
+
+# %%
+# We remove values on unstable terrain
+ddem.data[mask_glacier] = np.nan
+
+# %%
+# We use the reference DEM to derive terrain variables such as slope, aspect, curvature (see :ref:`sphx_glr_auto_examples_plot_terrain_attributes`)
+# that we'll use to explore potential non-stationarities in elevation measurement error
+
+# We compute the slope, aspect, and both plan and profile curvatures:
+slope, aspect, planc, profc = \
+    xdem.terrain.get_terrain_attribute(dem=ref_dem.data,
+                                       attribute=['slope','aspect', 'planform_curvature', 'profile_curvature'],
+                                       resolution=ref_dem.res)
+
+# %%
+# We use :func:`xdem.spstats.nd_binning` to perform N-dimensional binning on all those terrain variables, with uniform
+# bin length divided by 30. We use the :ref:`spatial_stats_nmad` as a robust measure of `statistical dispersion <https://en.wikipedia.org/wiki/Statistical_dispersion>`_.
+
+df = xdem.spstats.nd_binning(values=ddem.data,list_var=[slope, aspect, planc, profc],
+                             list_var_names=['slope','aspect','planc','profc'],
+                             statistics=['count',np.nanmedian,xdem.spstats.nmad],
+                             list_var_bins=30)
+
+# %%
+# We obtain a dataframe with the 1D binning results for each variable, the 2D binning results for all combinations of
```

Comment:
Panda's dataframe.
Is it possible to display the dataframe in the doc the same way it would show up in a notebook? Just try adding a print statement in the code.

Reply:
Yep! There's no super-pretty-print like in notebooks, but maybe the `df.to_markdown()` could render similarly?

Otherwise, just printing works.

Reply:
Turns out, `to_markdown()` needs an optional dependency. Maybe it could render html? `df.to_html()`

Reply:
Ah, you can just type the dataframe in the bottom of the cell, and it will show the html version (more specifically, it will call `__repl_html__()` if it exists.).

Reference: https://sphinx-gallery.github.io/stable/configuration.html?highlight=html#controlling-what-output-is-captured

Example:
![image](https://user-images.githubusercontent.com/33550973/124739102-d8d5d180-df19-11eb-9074-8c62221df51d.png)

Reply:
Leads me to think the `Raster` and `DEM` classes should have a `__repr_html__()` method!!

Reply:
Added https://github.com/GlacioHack/GeoUtils/issues/225

Reply:
Used this. With this method, the table will display as large as it needs to be, though:

![Screenshot from 2021-07-16 12-05-12](https://user-images.githubusercontent.com/28896516/125932054-3ee1de35-0bed-480f-8301-7a9f81598233.png)